### DISCUSSION TITLE
Encountering Error on autoreconf When Deploying OpenSSH on VM

### DISCUSSION THREAD

**Discussion by hongyucao** [open-quantum-safe/liboqs#1411] [Q&A] [answered]
Hello!

I'm a university student working on an undergraduate dissertation project about open quantum safe. I've hit some problems while trying to deploy OpenSSH on my virtual machine, and I'm hoping to get some help from this discussion area.
Error messages are as below during step 2 when I run build_openssh.sh:

\+ '[' -f Makefile ']'
\+ autoreconf -i
' is already registered with AC_CONFIG_FILES.
../../lib/autoconf/status.m4:288: AC_CONFIG_FILES is expanded from...
configure.ac:5622: the top level
autom4te: /usr/bin/m4 failed with exit status: 1
aclocal: error: echo failed with exit status: 1
autoreconf: aclocal failed with exit status: 1

My host machine is running Windows 10, and I'm using Vagrant on VMWare to run Ubuntu 20.04 on the virtual machine. I'm not sure how to resolve this problem, and I'm wondering if it might be a system conflict issue.

Any help or suggestions that you could provide would be greatly appreciated. Thanks in advance!

**Comment by baentsch** [comment, marked as answer]
There's lots of things that could have gone wrong. My strong suggestion is to tie down the environment (and the different packages and their versions used in build) to the max. You can do this for example by running the build inside of a docker image (using "openquantumsafe/ci-ubuntu-focal-x86_64" basically guarantees everything builds as we use that in CI).

**Reply by hongyucao** [reply]
Thank you very much, it took some time to set up then the script ran smoothly in this container environment!
